spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnector.scala

```diff
@@ -118,9 +118,10 @@ class CassandraConnector(conf: CassandraConnectorConf)
   def closestLiveHost: Host = {
     withClusterDo { cluster =>
       LocalNodeFirstLoadBalancingPolicy
-        .sortNodesByProximityAndStatus(_config.hosts, cluster.getMetadata.getAllHosts.toSet)
+        .sortNodesByStatusAndProximity(_config.hosts, cluster.getMetadata.getAllHosts.toSet)
+        .filter(_.isUp)
```

Comment:
The down nodes are still in the list and this claims that it will find the closest Live host, so if all the nodes are down it will still return a value when I believe it should not.